fl4health/parameter_exchange/layer_exchanger.py

```diff
@@ -92,88 +91,76 @@ def pull_parameters(self, parameters: NDArrays, model: nn.Module, config: Option
         model.load_state_dict(current_state, strict=True)
 
 
-class NormDriftParameterExchanger(ParameterExchangerWithPacking[List[str]]):
+class DynamicLayerExchanger(PartialParameterExchanger[List[str]]):
     def __init__(
         self,
+        layer_selection_function: Callable[..., Tuple[NDArrays, List[str]]],
         norm_threshold: float,
         exchange_percentage: float = 0.1,
         normalize: bool = True,
         filter_by_percentage: bool = True,
+        select_drift_more: bool = True,
     ) -> None:
         """
         This exchanger selects those parameters that at the end of each training round, drift away (in l2 norm)
         from their initial values (at the beginning of the same round) by more than self.threshold.
         """
+        self.set_layer_selection_function(layer_selection_function)
         self.parameter_packer = ParameterPackerWithLayerNames()
         assert 0 < exchange_percentage <= 1
         assert 0 < norm_threshold
-        self.threshold = norm_threshold
+        self.norm_threshold = norm_threshold
         self.exchange_percentage = exchange_percentage
         self.set_normalization_mode(normalize)
         self.set_filter_mode(filter_by_percentage)
+        self.set_select_drift_more(select_drift_more)
+
+    def set_layer_selection_function(
```

Comment:
Any reason that we have this set function rather than just doing `self.layer_selection_function = layer_selection_function` in the `__init__`? I know we use it below in the tests, but there we could simply do
```
exchanger.layer_selection_function = layer_selection_function
```
as well, since it's not a private attribute.

Reply:
I included these setters to give the user the ability to reset the selection criterion etc during training, but it's possible the situation where this is needed doesn't arise in practice... Currently they are only used in the tests. We can remove them if you think that is better.